### PR TITLE
fix: restore original /nix/store/ paths for buildenv

### PR DESCRIPTION
**Pull request by aakropotkin**
- fix: restore original buildenv store paths

When migrating `env-builder` into `pkgdb` I changed how `etc/profile.d/` and
`setup-prompt-bash.sh` were pulled into the executable by referring to them at
installed file-paths under `libexec/` and `share/`.

This was largely based on a misunderstanding of the existing usage because I
deleted `pkgs/flox-env-builder/default.nix` before
migrating `env-builder/src/flox-env.cc`.

This restores the original handling in the `nix` derivation, with an equivalent
fallback in the `Makefile`.

### CODE REVIEW

File: pkgdb/Makefile [hunk 1]
```diff
@@ -146,8 +142,17 @@ endif  # ifndef TOOLCHAIN
...
 # ---------------------------------------------------------------------------- #
 
+# Read info from files
+# --------------------
+
 VERSION := $(file < $(PKGDB_ROOT)/version)
 
+# Files used by `pkgdb buildenv' need to be added to the `nix' store.
+PROFILE_D_SCRIPT_DIR ?= $(shell                         \
+  $(NIX) store add-path -n profile.d src/buildenv/etc)
```
Comment (mkenigs) [inline pkgdb/Makefile:152]: nonblocking: iss the fallback for development? Could we fall back to the actual file then?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# ============================================================================ #
```
Comment (mkenigs) [inline Makefile:1]: Did you mean to commit this in this PR?

**Review by mkenigs — APPROVED** [review]

File: pkgdb/Makefile [hunk 3]
```diff
@@ -65,10 +63,8 @@ REPO_ROOT    := $(patsubst %/,%,$(dir $(MAKEFILE_DIR)))
 # Install Prefixes
 # ----------------
 
-PREFIX     ?= $(PKGDB_ROOT)/build
-BINDIR     ?= $(PREFIX)/bin
-LIBEXECDIR ?= $(PREFIX)/libexec
-SYSCONFDIR ?= $(PREFIX)/etc
+PREFIX ?= $(PKGDB_ROOT)/build
+BINDIR ?= $(PREFIX)/bin
```
Comment (ysndr) [inline pkgdb/Makefile:67]: It still confuses me that judging from these lines we build our binaries into a `build/bin` directory, but actually it still ends up in `/bin` or `/out/bin` if we run `make install`.

File: pkgdb/Makefile (hunk 1 above)
Comment (aakropotkin) [reply pkgdb/Makefile]: This is set as a fallback.
The path must be a `nix` store path otherwise it'll throw an error AFAIK.

File: Makefile (hunk 2 above)
Comment (aakropotkin) [reply Makefile:1]: Yes at the time.
Until this morning we had decided on using a top level `Makefile`.

Demo time was contradicted that agreement, so I can remove it until we settle that.

File: pkgdb/Makefile (hunk 3 above)
Comment (aakropotkin) [reply pkgdb/Makefile:67]: `PREFIX` is set by `nix` and represents the _install prefix_.

That directory will not appear locally anymore though after this merges.

**Review by zmitchell — APPROVED** [review]

**Review by mkenigs — APPROVED** [review]

**Review by garbas — APPROVED** [review]

**Review by mkenigs — APPROVED** [review]

**Comment by aakropotkin** [comment]
For whatever reason I can't attach this issue : https://github.com/flox/flox/issues/504

**Pull request merged**